### PR TITLE
[terra-worklist-data-grid] Add WDIO tests for pinned columns

### PR DESCRIPTION
**Pull request by sycombs**
### Summary

This PR adds WDIO tests for Worklist Data Grid pinned columns in order to validate pinned columns.

### Testing

This change was tested using:

- [x] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9437 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-worklist-data-grid/tests/wdio/worklist-data-grid-spec.js [hunk 1]
```diff
@@ -397,4 +397,33 @@ Terra.describeViewports('WorklistDataGrid', ['medium', 'large'], () => {
...
+    });
+
+    it('keeps row selection column stickied when scrolling', () => {
+      browser.url('/raw/tests/cerner-terra-framework-docs/worklist-data-grid/pinned-columns-with-row-selection');
+      navigateToCell(2, 1);
+      moveCurrentPositionBy(0, 9);
+      Terra.validates.element('pinned-columns-with-row-selection-scroll', { selector });
+    });
+
+    it('selects the whole row with pinned columns', () => {
```
Comment (nikhitasharma) [inline packages/terra-worklist-data-grid/tests/wdio/worklist-data-grid-spec.js:422]: I am having a hard time understanding the purpose of this test. It says `selects the whole row` but I do not see that selection on the screenshot ( I only see one cell selected) 
Am I missing something.

Comment (sycombs) [reply packages/terra-worklist-data-grid/tests/wdio/worklist-data-grid-spec.js:422]: Good catch, looks like the test file was not setting `hasSelectableRows` correctly. The screenshots should be fixed now.

File: packages/terra-framework-docs/src/terra-dev-site/test/worklist-data-grid/PinnedColumns.test.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  ],
+};
+
+const PinnedColumns = () => {
+  const rowHeaderIndex = 0;
+  const { cols, rows } = gridDataJSON;
+
+  return (
+    <WorklistDataGrid
+      id="default-terra-worklist-data-grid"
```
Comment (cm9361) [inline packages/terra-framework-docs/src/terra-dev-site/test/worklist-data-grid/PinnedColumns.test.jsx:60]: This test should have its own unique id for the WorklistDataGrid.

File: packages/terra-framework-docs/src/terra-dev-site/test/worklist-data-grid/PinnedColumns.test.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    { id: 'Column-1', displayName: 'Location' },
+    { id: 'Column-2', displayName: 'Illness Severity' },
+    { id: 'Column-3', displayName: 'Visit' },
+    { id: 'Column-4', displayName: 'Allergy' },
+    { id: 'Column-5', displayName: 'Primary Contact' },
+    { id: 'Column-6', displayName: 'Generic Order Counts' },
+    { id: 'Column-7', displayName: 'Patient Age' },
+    { id: 'Column-8', displayName: 'Medication History' },
+    { id: 'Column-9', displayName: 'My Relationship' },
+    { id: 'Column-10', displayName: 'Not Selectable', isSelectable: false },
```
Comment (cm9361) [inline packages/terra-framework-docs/src/terra-dev-site/test/worklist-data-grid/PinnedColumns.test.jsx:16]: Do we really need 11 columns?  It seems like we should be able to test pinned columns with like 4 or 5 columns of data.

File: packages/terra-framework-docs/src/terra-dev-site/test/worklist-data-grid/PinnedColumnsWithRowSelection.test.jsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        { content: '' },
+        { isMasked: true },
+        { content: 'Admitting Physician' },
+        { content: '', isSelectable: false },
+      ],
+    },
+  ],
+};
+
+const RowSelection = () => {
```
Comment (cm9361) [inline packages/terra-framework-docs/src/terra-dev-site/test/worklist-data-grid/PinnedColumnsWithRowSelection.test.jsx:54]: Why not just update the RowSelection test?  It doesn't feel like we should need two separate tests.

File: packages/terra-worklist-data-grid/tests/wdio/__snapshots__/reference/orion-fusion-theme/en/chrome_medium/worklist-data-grid-spec/pinned-columns-with-row-selection-scroll.png
Comment (nikhitasharma) [inline packages/terra-worklist-data-grid/tests/wdio/__snapshots__/reference/orion-fusion-theme/en/chrome_medium/worklist-data-grid-spec/pinned-columns-with-row-selection-scroll.png:1]: Is the focus here expected to overflow into the left pinned column or is it just the scrolling effecting the screenshot?

Comment (mjpalazzo) [reply packages/terra-worklist-data-grid/tests/wdio/__snapshots__/reference/orion-fusion-theme/en/chrome_medium/worklist-data-grid-spec/pinned-columns-with-row-selection-scroll.png:1]: Just the scrolling.

**Comment by sycombs** [comment]
> @sycombs - The pinned columns look good. I discussed an keyboard navigation issue with @KoushikKalli earlier today regarding scrolling. If I navigate all the way to the right using Right Arrow and then navigate back to the left using Left Arrow, some columns are hidden behind the pinned columns. There is a scroll bar below the grid that I can use with the mouse to navigate further and display these cells, but I cannot access the scroll bar with the keyboard. image

@mjpalazzo I believe this is something @sdadn discussed last week with @eawww (correct me if I'm wrong), and UXPLATFORM-9438 has been logged to address the scrolling issues.

File: packages/terra-framework-docs/src/terra-dev-site/test/worklist-data-grid/PinnedColumns.test.jsx (hunk 2 above)
Comment (sycombs) [reply packages/terra-framework-docs/src/terra-dev-site/test/worklist-data-grid/PinnedColumns.test.jsx]: Updated: [a7d0a64](https://github.com/cerner/terra-framework/pull/1736/commits/a7d0a648f4542468bbda83b8a82a6017b32c5fb5)

File: packages/terra-framework-docs/src/terra-dev-site/test/worklist-data-grid/PinnedColumnsWithRowSelection.test.jsx (hunk 4 above)
Comment (sycombs) [reply packages/terra-framework-docs/src/terra-dev-site/test/worklist-data-grid/PinnedColumnsWithRowSelection.test.jsx:54]: This test is specifically for the row selection case with additional pinned columns (other than the row selection column itself) to verify that row selection works correctly with pinned columns. If we combine the pinned columns case into the row selection tests, that distinction gets lost.

File: packages/terra-framework-docs/src/terra-dev-site/test/worklist-data-grid/PinnedColumns.test.jsx (hunk 3 above)
Comment (sycombs) [reply packages/terra-framework-docs/src/terra-dev-site/test/worklist-data-grid/PinnedColumns.test.jsx:16]: With 4 or 5 columns scrolling will not be present/as apparent on larger screen sizes, the greater number of columns makes it easier to validate the scroll behavior.

**Review by cm9361 — APPROVED** [review]

**Pull request merged**